pkg/apiserver/apiserver.go

```diff
@@ -36,49 +38,133 @@ import (
 	"github.com/pingcap-incubator/tidb-dashboard/pkg/apiserver/utils"
 	"github.com/pingcap-incubator/tidb-dashboard/pkg/config"
 	"github.com/pingcap-incubator/tidb-dashboard/pkg/dbstore"
+	http2 "github.com/pingcap-incubator/tidb-dashboard/pkg/http"
 	"github.com/pingcap-incubator/tidb-dashboard/pkg/keyvisual"
+	keyvisualregion "github.com/pingcap-incubator/tidb-dashboard/pkg/keyvisual/region"
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/pd"
 	"github.com/pingcap-incubator/tidb-dashboard/pkg/tidb"
+	utils2 "github.com/pingcap-incubator/tidb-dashboard/pkg/utils"
+)
+
+var (
+	once sync.Once
 )
 
-var once sync.Once
+type PDDataProviderConstructor func(*config.Config, *http.Client, *clientv3.Client) *keyvisualregion.PDDataProvider
 
-type Services struct {
-	Store         *dbstore.DB
-	TiDBForwarder *tidb.Forwarder
-	KeyVisual     *keyvisual.Service
-	EtcdProvider  pd.EtcdProvider
-	HTTPClient    *http.Client
+type Service struct {
+	app    *fx.App
+	status *utils2.ServiceStatus
```

Comment:
`runningStatus *utils2.ServiceRunningStatus`

Reply:
It looks like `Running` should be an enum of `ServiceStatus`